lib/srv/db/cloud/resource_checker_url_aws.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/opensearchservice"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/aws/aws-sdk-go/service/redshiftserverless/redshiftserverlessiface"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	apiawsutils "github.com/gravitational/teleport/api/utils/aws"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+func (c *urlChecker) checkAWS(describeCheck, basicEndpointCheck checkDatabaseFunc) checkDatabaseFunc {
+	return func(ctx context.Context, database types.Database) error {
+		err := describeCheck(ctx, database)
+
+		// Database Service may not have enough permissions to permform the
+		// describes. Log a warning and permform a basic endpoint validation
+		// instead.
+		if trace.IsAccessDenied(err) {
+			c.logAWSAccessDeniedError(database, err)
+
+			if err := basicEndpointCheck(ctx, database); err != nil {
+				return trace.Wrap(err)
+			}
+			c.log.Debugf("Database %v URL validated by basic endpoint check.", database.GetName())
+			return nil
+		}
+
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		c.log.Debugf("Database %v URL validated by describe check.", database.GetName())
+		return nil
+	}
+}
+
+func (c *urlChecker) logAWSAccessDeniedError(database types.Database, accessDeinedError error) {
+	c.warnAWSOnce.Do(func() {
+		// TODO(greedy52) add links to doc.
+		c.log.Warn("No permissions to describe AWS resource metadata that is needed for validating databases created by Discovery Service. Basic AWS endpoint validation will be performed instead. For best security, please provide the Database Service with the proper IAM permissions. Enable --debug mode to see details on which databases require more IAM permissions. See Database Access documentation for more details.")
+	})
+
+	c.log.Debugf("No permissions to describe %q: %v.", database, accessDeinedError)
```

Comment:
Can we add more context: like Database URL verification failed due to missing AWS IAM permission. 

Also printing whole `database` object with  `accessDeinedError` will produce log noise. If we know that accessDeinedError == aws access denied error we can skip it from debug trace.